man/man3/conf.3
.TH "conf" 3 "Sat Aug 8 2020" "libcp2p" \" -*- nroff -*-
.ad l
.nh
.SH NAME
conf
.SH SYNOPSIS
.br
.PP
.SS "Variables"

.in +1c
.ti -1c
.RI "list \fBextensions\fP = []"
.br
.ti -1c
.RI "list \fBtemplates_path\fP = ['_templates']"
.br
.ti -1c
.RI "string \fBsource_suffix\fP = '\&.rst'"
.br
.ti -1c
.RI "string \fBmaster_doc\fP = 'index'"
.br
.ti -1c
.RI "string \fBproject\fP = 'c\-template\-REPLACEME'"
.br
.ti -1c
.RI "string \fBcopyright\fP = '2020 c\-template\-REPLACEME'"
.br
.ti -1c
.RI "string \fBauthor\fP = 'c\-template\-REPLACEME'"
.br
.ti -1c
.RI "string \fBversion\fP = ''"
.br
.ti -1c
.RI "string \fBrelease\fP = ''"
.br
.ti -1c
.RI "\fBlanguage\fP = None"
.br
.ti -1c
.RI "list \fBexclude_patterns\fP = ['_build', 'Thumbs\&.db', '\&.DS_Store']"
.br
.ti -1c
.RI "string \fBpygments_style\fP = 'sphinx'"
.br
.ti -1c
.RI "bool \fBtodo_include_todos\fP = False"
.br
.ti -1c
.RI "string \fBhtml_theme\fP = 'alabaster'"
.br
.ti -1c
.RI "list \fBhtml_static_path\fP = []"
.br
.ti -1c
.RI "dictionary \fBhtml_sidebars\fP"
.br
.ti -1c
.RI "string \fBhtmlhelp_basename\fP = 'c\-template\-REPLACEME\-doc'"
.br
.ti -1c
.RI "dictionary \fBlatex_elements\fP"
.br
.ti -1c
.RI "list \fBlatex_documents\fP"
.br
.ti -1c
.RI "list \fBman_pages\fP = [(\fBmaster_doc\fP, 'c\-template\-REPLACEME', 'c\-template\-REPLACEME Documentation', [\fBauthor\fP], 1)]"
.br
.in -1c
.SH "Variable Documentation"
.PP 
.SS "string conf\&.author = 'c\-template\-REPLACEME'"

.PP
Definition at line 50 of file conf\&.py\&.
.SS "string conf\&.copyright = '2020 c\-template\-REPLACEME'"

.PP
Definition at line 49 of file conf\&.py\&.
.SS "list conf\&.exclude_patterns = ['_build', 'Thumbs\&.db', '\&.DS_Store']"

.PP
Definition at line 71 of file conf\&.py\&.
.SS "list conf\&.extensions = []"

.PP
Definition at line 33 of file conf\&.py\&.
.SS "dictionary conf\&.html_sidebars"
\fBInitial value:\fP
.PP
.nf
1 =  {
2     '**': [
3         'relations\&.html',  # needs 'show_related': True theme option to display
4         'searchbox\&.html',
5     ]
6 }
.fi
.PP
Definition at line 102 of file conf\&.py\&.
.SS "list conf\&.html_static_path = []"

.PP
Definition at line 95 of file conf\&.py\&.
.SS "string conf\&.html_theme = 'alabaster'"

.PP
Definition at line 84 of file conf\&.py\&.
.SS "string conf\&.htmlhelp_basename = 'c\-template\-REPLACEME\-doc'"

.PP
Definition at line 112 of file conf\&.py\&.
.SS "conf\&.language = None"

.PP
Definition at line 66 of file conf\&.py\&.
.SS "list conf\&.latex_documents"
\fBInitial value:\fP
.PP
.nf
1 =  [
2     (master_doc, 'c-template-REPLACEME\&.tex', 'c-template-REPLACEME Documentation',
3      'c-template-REPLACEME\&. c-template-AUTHOR', 'manual'),
4 ]
.fi
.PP
Definition at line 137 of file conf\&.py\&.
.SS "dictionary conf\&.latex_elements"
\fBInitial value:\fP
.PP
.nf
1 =  {
2     # The paper size ('letterpaper' or 'a4paper')\&.
3     #
4     # 'papersize': 'letterpaper',
5 
6     # The font size ('10pt', '11pt' or '12pt')\&.
7     #
8     # 'pointsize': '10pt',
9 
10     # Additional stuff for the LaTeX preamble\&.
11     #
12     # 'preamble': '',
13 
14     # Latex figure (float) alignment
15     #
16     # 'figure_align': 'htbp',
17 }
.fi
.PP
Definition at line 116 of file conf\&.py\&.
.SS "list conf\&.man_pages = [(\fBmaster_doc\fP, 'c\-template\-REPLACEME', 'c\-template\-REPLACEME Documentation', [\fBauthor\fP], 1)]"

.PP
Definition at line 146 of file conf\&.py\&.
.SS "string conf\&.master_doc = 'index'"

.PP
Definition at line 45 of file conf\&.py\&.
.SS "string conf\&.project = 'c\-template\-REPLACEME'"

.PP
Definition at line 48 of file conf\&.py\&.
.SS "string conf\&.pygments_style = 'sphinx'"

.PP
Definition at line 74 of file conf\&.py\&.
.SS "string conf\&.release = ''"

.PP
Definition at line 59 of file conf\&.py\&.
.SS "string conf\&.source_suffix = '\&.rst'"

.PP
Definition at line 42 of file conf\&.py\&.
.SS "list conf\&.templates_path = ['_templates']"

.PP
Definition at line 36 of file conf\&.py\&.
.SS "bool conf\&.todo_include_todos = False"

.PP
Definition at line 77 of file conf\&.py\&.
.SS "string conf\&.version = ''"

.PP
Definition at line 57 of file conf\&.py\&.
.SH "Author"
.PP 
Generated automatically by Doxygen for libcp2p from the source code\&.


docs/conf.py
#!@PYTHON_EXECUTABLE@
# -*- coding: utf-8 -*-
#
# recipe-02 documentation build configuration file, created by
# sphinx-quickstart on Thu Feb 15 17:02:41 2018.
#
# This file is execfile()d with the current directory set to its
# containing dir.
#
# Note that not all possible configuration values are present in this
# autogenerated file.
#
# All configuration values have a default; values that are commented out
# serve to show the default.

# If extensions (or modules to document with autodoc) are in another directory,
# add these directories to sys.path here. If the directory is relative to the
# documentation root, use os.path.abspath to make it absolute, like shown here.
#
# import os
# import sys
# sys.path.insert(0, os.path.abspath('.'))

# -- General configuration ------------------------------------------------

# If your documentation needs a minimal Sphinx version, state it here.
#
# needs_sphinx = '1.0'

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
# ones.
extensions = []

# Add any paths that contain templates here, relative to this directory.
templates_path = ['_templates']

# The suffix(es) of source filenames.
# You can specify multiple suffix as a list of string:
#
# source_suffix = ['.rst', '.md']
source_suffix = '.rst'

# The master toctree document.
master_doc = 'index'

# General information about the project.
project = 'c-template-REPLACEME'
copyright = '2020 c-template-REPLACEME'
author = 'c-template-REPLACEME'

# The version info for the project you're documenting, acts as replacement for
# |version| and |release|, also used in various other places throughout the
# built documents.
#
# The short X.Y version.
version = ''
# The full version, including alpha/beta/rc tags.
release = ''

# The language for content autogenerated by Sphinx. Refer to documentation
# for a list of supported languages.
#
# This is also used if you do content translation via gettext catalogs.
# Usually you set "language" from the command line for these cases.
language = None

# List of patterns, relative to source directory, that match files and
# directories to ignore when looking for source files.
# This patterns also effect to html_static_path and html_extra_path
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']

# The name of the Pygments (syntax highlighting) style to use.
pygments_style = 'sphinx'

# If true, `todo` and `todoList` produce output, else they produce nothing.
todo_include_todos = False

# -- Options for HTML output ----------------------------------------------

# The theme to use for HTML and HTML Help pages.  See the documentation for
# a list of builtin themes.
#
html_theme = 'alabaster'

# Theme options are theme-specific and customize the look and feel of a theme
# further.  For a list of options available for each theme, see the
# documentation.
#
# html_theme_options = {}

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = []

# Custom sidebar templates, must be a dictionary that maps document names
# to template names.
#
# This is required for the alabaster theme
# refs: http://alabaster.readthedocs.io/en/latest/installation.html#sidebars
html_sidebars = {
    '**': [
        'relations.html',  # needs 'show_related': True theme option to display
        'searchbox.html',
    ]
}

# -- Options for HTMLHelp output ------------------------------------------

# Output file base name for HTML help builder.
htmlhelp_basename = 'c-template-REPLACEME-doc'

# -- Options for LaTeX output ---------------------------------------------

latex_elements = {
    # The paper size ('letterpaper' or 'a4paper').
    #
    # 'papersize': 'letterpaper',

    # The font size ('10pt', '11pt' or '12pt').
    #
    # 'pointsize': '10pt',

    # Additional stuff for the LaTeX preamble.
    #
    # 'preamble': '',

    # Latex figure (float) alignment
    #
    # 'figure_align': 'htbp',
}

# Grouping the document tree into LaTeX files. List of tuples
# (source start file, target name, title,
#  author, documentclass [howto, manual, or own class]).
latex_documents = [
    (master_doc, 'c-template-REPLACEME.tex', 'c-template-REPLACEME Documentation',
     'c-template-REPLACEME. c-template-AUTHOR', 'manual'),
]

# -- Options for manual page output ---------------------------------------

# One entry per manual page. List of tuples
# (source start file, name, description, authors, manual section).
man_pages = [(master_doc, 'c-template-REPLACEME', 'c-template-REPLACEME Documentation', [author], 1)]

# -- Options for Texinfo output -------------------------------------------

# Grouping the document tree into Texinfo files. List of tuples
# (source start file, target name, title, author,
#  dir menu entry, description, category)
texinfo_documents = [
    (master_doc, 'c-template-REPLACEME', 'c-template-REPLACEME Documentation', author, 'c-template-REPLACEME',
     'One line description of project.', 'Miscellaneous'),
]